Detectors/MUON/MCH/Align/src/AlignRecordSpec.cxx

```diff
@@ -0,0 +1,369 @@
+      mAlign.FixChamber(chamber);
+    }
+
+    // Init for output saving
+    auto OutputRecFileName = ic.options().get<string>("output-record-data");
+    auto OutputConsFileName = ic.options().get<string>("output-record-constraint");
+    mAlign.init(OutputRecFileName, OutputConsFileName);
+
+    ic.services().get<CallbackService>().set<CallbackService::Id::Stop>([this]() {
+      LOG(info) << "Saving records into ROOT file";
+      LOG(info) << "Nb of records being saved: " << mTrackCount;
+      LOG(info) << "Nb of matched MCH-MID tracks: " << mTrackMatched;
+      LOG(info) << "Nb of Matched MCH-MID tracks actually used: " << mTrackMatchedPerTF;
+      mAlign.terminate();
+    });
+  }
+
+  //_________________________________________________________________________________________________
+  void finaliseCCDB(framework::ConcreteDataMatcher& matcher, void* obj)
+  {
+    /// finalize the track extrapolation setting
+    if (mCCDBRequest && base::GRPGeomHelper::instance().finaliseCCDB(matcher, obj)) {
+      if (matcher == framework::ConcreteDataMatcher("GLO", "GRPMAGFIELD", 0)) {
+        TrackExtrap::setField();
+        TrackExtrap::useExtrapV2();
+        mAlign.SetBFieldOn(mch::TrackExtrap::isFieldON());
+      }
+
+      if (matcher == framework::ConcreteDataMatcher("GLO", "GEOMALIGN", 0)) {
+        LOG(info) << "Loading reference geometry from CCDB";
+        transformation = geo::transformationFromTGeoManager(*gGeoManager);
+        for (int i = 0; i < 156; i++) {
+          int iDEN = GetDetElemId(i);
+          transform[iDEN] = transformation(iDEN);
+        }
+      }
+    }
+  }
+
+  //_________________________________________________________________________________________________
+  void run(framework::ProcessingContext& pc)
+  {
+    if (mCCDBRequest) {
+      base::GRPGeomHelper::instance().checkUpdates(pc);
+    }
+
+    o2::globaltracking::RecoContainer recoData;
+    recoData.collectData(pc, *mDataRequest.get());
+
+    const auto& mchTracks = recoData.getMCHTracks();
+    const auto& mchmidMatches = recoData.getMCHMIDMatches();
+    const auto& mchClusters = recoData.getMCHTrackClusters();
+    mTrackMatched += mchmidMatches.size();
+
+    for (auto const& mchmidMatch : mchmidMatches) {
+
+      int mchTrackID = mchmidMatch.getMCHRef().getIndex();
+      const auto& mchTrack = mchTracks[mchTrackID];
+      int first = mchTrack.getFirstClusterIdx();
+      int last = mchTrack.getLastClusterIdx();
+      int Ncluster = last - first + 1;
+
+      if (Ncluster <= 9) {
+        continue;
+      }
+
+      mch::Track convertedTrack;
+
+      for (int i = first; i <= last; i++) {
+        const auto& cluster = mchClusters[i];
+        convertedTrack.createParamAtCluster(cluster);
+      }
+
+      // Erase removable track
+      if (!RemoveTrack(convertedTrack)) {
+        mAlign.ProcessTrack(convertedTrack, transformation, false, weightRecord);
+        mTrackCount += 1;
+        mTrackMatchedPerTF += 1;
```

Comment:
this is a detail, but currently `mTrackCount` and `mTrackMatchedPerTF` counters are identical.

Reply:
Yes, indeed, another one was meant to check per TF track candidates, then I made some changes; I should have removed it.